pkg/agent/controller/agent.go

```diff
@@ -288,10 +375,67 @@ func (a *Controller) newServiceImport(svcExport *lighthousev2a1.ServiceExport) *
 				originName:      svcExport.Name,
 				originNamespace: svcExport.Namespace,
 			},
+			Labels: map[string]string{
+				labelSourceName:      svcExport.Name,
+				labelSourceNamespace: svcExport.Namespace,
+				labelSourceCluster:   a.clusterID,
+			},
 		},
 	}
 }
 
+func (a *Controller) getServiceImport(svcExport *lighthousev2a1.ServiceExport) (*lighthousev2a1.ServiceImport, error) {
+	labelMap := map[string]string{
+		labelSourceName:      svcExport.Name,
+		labelSourceNamespace: svcExport.Namespace,
+		labelSourceCluster:   a.clusterID,
+	}
+	siListOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelMap).String(),
+	}
+	serviceImports, err := a.lighthouseClient.LighthouseV2alpha1().ServiceImports(corev1.NamespaceAll).List(siListOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(serviceImports.Items) > 1 {
+		return nil, fmt.Errorf("Found multiple ServiceImports with local cluster ID.")
+	}
+
+	return &serviceImports.Items[0], nil
```

Comment:
`serviceImports.Items` could be empty.

But instead of using a label selector for lookup, why not  lookup by Name as constructed on L373 (and put it in a `serviceNameFrom(svcExport *lighthousev2a1.ServiceExport)` function)? The same label fields you're using for lookup are also encoded in the Name. This code would be simplified to use `Get` and not have to check for multiple.